File: 11. Maximum Score After Splitting a String.py
class Solution(object):
    def maxScore(self, s):
        """
        :type s: str
        :rtype: int
        """
        # Calculate the total number of ones in the string
        total_ones = s.count('1')
        
        # Initialize variables to keep track of the left zeros, right ones, and max score
        left_zeros = 0
        right_ones = total_ones
        max_score = 0
        
        # Iterate through the string, splitting it into two non-empty substrings
        for i in range(len(s) - 1):  # Exclude the last character to keep both substrings non-empty
            if s[i] == '0':
                left_zeros += 1
            else:
                right_ones -= 1
            
            # Calculate the score for the current split
            current_score = left_zeros + right_ones
            max_score = max(max_score, current_score)
        
        return max_score

File: 11. Maximum Score After Splitting a String.java
class Solution {
    public int maxScore(String s) {
        // Calculate the total number of ones in the string
        int totalOnes = 0;
        for (char c : s.toCharArray()) {
            if (c == '1') {
                totalOnes++;
            }
        }

        // Initialize variables to keep track of the left zeros, right ones, and max score
        int leftZeros = 0;
        int rightOnes = totalOnes;
        int maxScore = 0;

        // Iterate through the string, splitting it to two non-empty substrings
        for (int i = 0; i < s.length() - 1; i++) { // Exclude the last character to keep both substrings non-empty
            if (s.charAt(i) == '0') {
                leftZeros++;
            } else {
                rightOnes--;
            }

            // Calculate the score for the current split
            int currentScore = leftZeros + rightOnes;
            maxScore = Math.max(maxScore, currentScore);
        }

        return maxScore;
    }
}
